[MTM-65709] Documented trusted certificate renewal feature (#4076)
SoftwareAG/c8y-docs: content/device-certificate-authentication/managing-trusted-certificates.md

For details on the fields, see the description on adding certificates above.

> ### To renew a trusted certificate {#to-renew-a-trusted-certificate}

Reviewer: I think we should emphasize that this is only about the Cumulocity certificate, not all trusted certificates

Author: ```suggestion
### To renew a CA certificate {#to-renew-a-ca-certificate}
```

> Certificate renewal enables you to generate a new Certificate Authority (CA) certificate before the existing one expires. This maintains uninterrupted device connectivity and prevents authentication failures when the current CA certificate reaches the end of its validity period.

Reviewer: I think it's worth adding information here explaining that the CA certificate is "provided by Cumulocity". Because theoretically, all certificates are CA, but we can only refresh the one we provided. And regarding 'uninterrupted device connectivity,' it just turned out (L2S-1346) that refreshing the certificate disconnects all devices connected by certificates signed by our CA. So I'd prefer not to use this word:  "uninterrupted device connectivity."

Author: I see, how about:

> ### To renew a CA certificate provided by {{< company-c8y >}} {#to-renew-a-ca-certificate}
> 
> Certificate renewal enables you to generate a new Certificate Authority (CA) certificate provided by {{< company-c8y >}} before the existing one expires. The connected devices may need a moment to reconnect after the renewal, but it prevents sudden authentication failures in case the current certificate reaches the end of its validity period.

> Certificate renewal enables you to generate a new Certificate Authority (CA) certificate before the existing one expires. This maintains uninterrupted device connectivity and prevents authentication failures when the current CA certificate reaches the end of its validity period.

Author: ```suggestion
### To renew a CA certificate provided by {{< company-c8y >}} {#to-renew-a-ca-certificate}

Certificate renewal enables you to generate a new Certificate Authority (CA) certificate provided by {{< company-c8y >}} before the existing one expires. The connected devices may need a moment to reconnect after the renewal, but it prevents sudden authentication failures in case the current certificate reaches the end of its validity period.
```

> Certificate renewal enables you to generate a new certificate authority (CA) certificate before the existing one expires. This maintains uninterrupted device connectivity and prevents authentication failures when the current CA certificate reaches the end of its validity period.

Reviewer 2: ```suggestion
Certificate renewal enables you to generate a new Certificate Authority (CA) certificate before the existing one expires. This maintains uninterrupted device connectivity and prevents authentication failures when the current CA certificate reaches the end of its validity period.
```

> A **Renew certificate** button appears for a certificate in the certificate list view when the following conditions are met:

Reviewer 2: ```suggestion
A **Renew certificate** button appears for a certificate in the certificate list when the following conditions are met:
```

* The certificate is a CA certificate.

> * The certificate expires within 1.5 years from the current date.

Reviewer 3: I guess period is 3 years for CA expire. @mbak-c8y please correct me if I am wrong.

Author: Note that it refers to when you can renew a certificate, not how long the certificate is valid overall, according to this comment https://cumulocity.atlassian.net/browse/MTM-62992?focusedCommentId=1280917, the renewal is only possible 18 months before the certificate expires.

Reviewer 3: @pawel-rynarzewski-c8y correct

Reviewer: Actually com.cumulocity.model.certificate.X509DeviceCertificate#EXPIRY_THRESHOLD_MONTHS
is set to 24 months.
it was changed: https://cumulocity.atlassian.net/browse/MTM-64489

Author: I see, so I need to update it on UI side, too. Or, can we read that value via API?

Reviewer: It is static for all env. Can be hardcoded

Author: ```suggestion
* The certificate expires within 2 years from the current date.
```

To renew a CA certificate perform these steps:

1. In the **Trusted certificates** list, find the certificate you want to renew.

> 2. Click the <i class="dlt-c8y-icon-refresh text-primary icon-20"></i> **Renew certificate** icon next to its current expiration date.

Reviewer 2: @pawel-rynarzewski-c8y Ia this an icon with a label? Otherwise the text should just generically be "refresh icon" (not formatted bold). Compare "delete icon" further below.

Author: Right, it's an icon with this text as tooltip. Aligned with "delete icon".
```suggestion
2. Click the refresh icon <i class="dlt-c8y-icon-refresh text-primary icon-20"></i> next to its current expiration date.
```
